zio-http/src/main/scala/zhttp/http/Request.scala

```diff
@@ -1,10 +1,13 @@
 package zhttp.http
 
+import java.net.InetAddress
+
 // REQUEST
 final case class Request(
   endpoint: Endpoint,
   headers: List[Header] = List.empty,
   content: HttpData[Any, Nothing] = HttpData.empty,
+  remote: Option[InetAddress] = None,
```

Comment:
Im not really sure about this API. 
There is going to be a performance penalty even in-case someone isn't reading the IP address.

Reply:
What performance penalty? You mean unnecessary allocation of `Option` instance?

Do you have other solution in mind? The API doesn't have to be like this, I just think that having access to remote address is crucial for certain use cases.

Reply:
One option is to save `context` inside the request and have a method `localAddress()` that returns the address from the context.

Reply:
Not sure whether it's much better but I've refactored it that way. Can you please look at it again? Thanks.